### DISCUSSION TITLE
Whole page gets refreshed when changing routes on rare occasions

### DISCUSSION THREAD

**Discussion by sjsakib** [vercel/next.js#31114] [Help]
When changing routes the whole page sometimes get's refreshed, causing our app state to be lost. It happens very rarely, like 1-2% of times, but it keeps happening. What could be causing it? Please help us understand if this is fixable, otherwise we'll have to persist our whole  app state.

**Comment by fa-901** [comment]
Have you used the `Link` component from `next/link` to handle route changes ?

**Reply by sjsakib** [reply]
Actually we are using a mix of  `router.push` and `Link` the problem seems to be happening in both cases
